packit_service/worker/build/build_helper.py

```diff
@@ -323,10 +324,34 @@ def get_test_check(cls, chroot: str = None) -> str:
         chroot_str = f"-{chroot}" if chroot else ""
         return f"{deployment_str}/{cls.status_name_test}{chroot_str}"
 
-    def create_srpm_if_needed(self) -> None:
-        """If you want to be sure we already created the SRPM."""
-        if not (self._srpm_path or self._srpm_model):
-            self._create_srpm()
+    def create_srpm_if_needed(self) -> Optional[TaskResults]:
+        """
+        Create SRPM if is needed.
+
+        Returns:
+            Task results if job is cancelled because of merge conflicts, `None`
+        otherwise.
+        """
+        if self._srpm_path or self._srpm_model:
+            return None
+
+        if pr_id := self.pr_id:
+            state = self.project.get_pr(pr_id).merge_commit_status
+
+            if state == MergeCommitStatus.cannot_be_merged:
+                self.report_status_to_all(
+                    description="Merge conflicts present",
+                    state=BaseCommitStatus.neutral,
+                )
+                return TaskResults(
+                    success=True,
+                    details={
+                        "msg": "Merge conflicts were detected, cannot build SRPM."
+                    },
+                )
+
+        self._create_srpm()
```

Comment:
This can still fail due to a race condition - time of check & time of use. So we should prepared here.

Reply:
you mean race condition between maintainers/CI and the „state“ from PR?